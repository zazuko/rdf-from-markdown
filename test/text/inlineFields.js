import { expect } from 'expect'

import toMatchSnapshot from 'expect-mocha-snapshot'
import {
  extractInlineFields,
} from '../../src/text/inlineFields.js'

const markdown = [
  'inline :: field',
  'nested ( inline :: field )',
  '**Thoughts**:: It was decent.',
  '**Rating**:: 6',
  '[mood:: okay] | [length:: 2 hours]']

expect.extend({ toMatchSnapshot })

describe('extractInlineFields', async function () {
  for (const current of markdown) {
    it(current, async function () {

      const fields = extractInlineFields(current)

      expect(fields).toMatchSnapshot(this)
    })
  }
})
