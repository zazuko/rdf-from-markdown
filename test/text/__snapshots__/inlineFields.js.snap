// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractInlineFields **Rating**:: 6 1`] = `
Object {
  "property": "**Rating**",
  "value": " 6",
}
`;

exports[`extractInlineFields **Thoughts**:: It was decent. 1`] = `
Object {
  "property": "**Thoughts**",
  "value": " It was decent.",
}
`;

exports[`extractInlineFields [mood:: okay] | [length:: 2 hours] 1`] = `
Array [
  Object {
    "property": "mood",
    "value": " okay",
  },
  Object {
    "property": "length",
    "value": " 2 hours",
  },
]
`;

exports[`extractInlineFields inline :: field 1`] = `
Object {
  "property": "inline ",
  "value": " field",
}
`;

exports[`extractInlineFields nested ( inline :: field ) 1`] = `
Array [
  Object {
    "property": " inline ",
    "value": " field ",
  },
]
`;
